packages/components/pie-cookie-banner/test/helpers/page-object/pie-cookie-banner.page.ts

```diff
@@ -0,0 +1,135 @@
+        this.bodyAcceptAllLinkLocator = page.getByTestId(cookieBanner.selectors.bodyAcceptAllLink.dataTestId);
+        this.bodyNecessaryOnlyLinkLocator = page.getByTestId(cookieBanner.selectors.bodyNecessaryOnlyLink.dataTestId);
+        this.bodyManagePreferencesLinkLocator = page.getByTestId(cookieBanner.selectors.bodyManagePreferencesLink.dataTestId);
+        this.bodyCookieStatementLinkLocator = page.getByTestId(cookieBanner.selectors.bodyCookieStatementLink.dataTestId);
+        this.bodyCookieTechnologiesLinkLocator = page.getByTestId(cookieBanner.selectors.bodyCookieTechnologiesLink.dataTestId);
+        this.modalDescriptionLocator = page.getByTestId(cookieBanner.selectors.modalDescription.dataTestId);
+    }
+
+    async isCookieBannerVisible () {
+        return this.componentLocator.isVisible();
+    }
+
+    async clickAcceptAllButton () {
+        await this.acceptAllButtonLocator.click();
+    }
+
+    async clickBodyAcceptAllLink () {
+        await this.bodyAcceptAllLinkLocator.click();
+    }
+
+    async clickAcceptAll (elementLevel: string) {
+        await (elementLevel === 'action' ? this.clickAcceptAllButton() : this.clickBodyAcceptAllLink());
+    }
+
+    async getAcceptAllAttribute (attribute: string) {
+        return this.acceptAllButtonLocator.getAttribute(attribute);
+    }
+
+    async clickNecessaryOnlyButton () {
+        await this.necessaryOnlyButtonLocator.click();
+    }
+
+    async getNecessaryOnlyButtonAttribute (attribute: string) {
+        return this.necessaryOnlyButtonLocator.getAttribute(attribute);
+    }
+
+    async clickNecessaryOnlyLink () {
+        await this.bodyNecessaryOnlyLinkLocator.click();
+    }
+
+    async clickNecessaryOnlyAll (elementLevel: string) {
+        await (elementLevel === 'action' ? this.clickNecessaryOnlyButton() : this.clickNecessaryOnlyLink());
+    }
+
+    async clickManagePreferencesAction () {
+        await this.managePrefsLinkLocator.click();
+    }
+
+    async clickManagePreferencesLink () {
+        await this.bodyManagePreferencesLinkLocator.click();
+    }
+
+    async clickManagePreferencesAll (elementLevel: string) {
+        await (elementLevel === 'action' ? this.clickManagePreferencesAction() : this.clickManagePreferencesLink());
+    }
+
+    async clickPreferenceToggle (preferenceIds: string) {
+        const preferenceToggle = this.page.locator(preferenceIds);
+        return preferenceToggle.click();
+    }
+
+    async isPreferenceToggleChecked (preferenceIds: string) {
+        const preferenceToggle = this.page.locator(preferenceIds);
+        return preferenceToggle.isChecked();
+    }
+
+    async isPreferenceToggleDisabled (preferenceIds: string) {
+        const preferenceToggle = this.page.locator(preferenceIds);
+        return preferenceToggle.isChecked();
+    }
+
+    async getAcceptAllTextContent () {
+        const acceptAllText = await this.acceptAllButtonLocator.textContent();
+        return String(acceptAllText).trim();
```

Comment:
Out of interest, why is `String` needed here?

Reply:
Just thinking that if `acceptAllText` comes back as `null`, this will return `"null"`.
Would this do what you need it to?
```suggestion
        return acceptAllText?.trim();
```
It returns either `undefined` or the trimmed string.